// loopd-logger/app.js
'use strict';
var fs = require('fs');
var bunyan = require('bunyan');
var bformat = require('bunyan-format');
var formatOut = bformat({
    outputMode: 'long'
});
// var bunyanLumberjackStream = require('bunyan-lumberjack');
var bunyanRequest = require('bunyan-request');
// var bunyantcp = require('bunyan-logstash-tcp');
// var ELK_LOG = process.env.ELK_LOG;

var logStreams = [{
    level: process.env.LOG_LEVEL || 'info',
    stream: formatOut
}];
// var tcpStream = {};

// if (!ELK_LOG) {
//     setTimeout(function(){
//         console.log('==============No need to export logs!==============');
//     }, 5000);
// } else {
//     tcpStream.level = process.env.LOG_LEVEL || 'info';
//     tcpStream.type = 'raw';
//     tcpStream.stream = bunyantcp.createStream({
//         host: 'logstash-shipper.loopd.com',
//         port: 9998
//     });
//     logStreams.push(tcpStream);
// }

bunyan.DISABLE = 70;

var logger = bunyan.createLogger({
    name: process.env.LOG_TAG || 'unknown',
    env: process.env.NODE_ENV || 'unknown',
    streams: logStreams
});

var logRequestMiddleware = bunyanRequest({
    logger: logger,
    headerName: 'x-request-id'
});

module.exports.logger = logger;
module.exports.logRequestMiddleware = logRequestMiddleware;
